monitoring/ecs_dashboard/update_service_list/src/update_service_list.py

```diff
@@ -26,58 +26,66 @@ def _create_event_dict(event):
     return {"timestamp": event["createdAt"].timestamp(), "message": event["message"]}
 
 
-def _create_cluster_dict(cluster, service_list):
-    return {
-        "clusterName": cluster["clusterName"],
-        "status": cluster["status"],
-        "instanceCount": cluster["registeredContainerInstancesCount"],
-        "serviceList": service_list,
-    }
+def _describe_task_definition(client, arn):
+    return client.describe_task_definition(taskDefinition=arn)["taskDefinition"]
 
 
-def _create_service_dict(service):
-    deployments = [_create_deployment_dict(d) for d in service["deployments"]]
-    # Grab just the last few events to keep the file size down
-    events = [_create_event_dict(e) for e in service["events"][:5]]
+def _enrich_deployment(deployment, task_definition):
+    task_definition_arn = deployment["taskDefinition"]
 
-    return {
-        "serviceName": service["serviceName"],
-        "desiredCount": service["desiredCount"],
-        "pendingCount": service["pendingCount"],
-        "runningCount": service["runningCount"],
-        "deployments": deployments,
-        "events": events,
-        "status": service["status"],
-    }
+    task_definition["taskDefinitionArn"] = task_definition_arn
+    deployment["taskDefinition"] = task_definition
 
+    return deployment
 
-def _create_deployment_dict(deployment):
-    return {
-        "id": deployment["id"],
-        "taskDefinition": deployment["taskDefinition"],
-        "status": deployment["status"],
-    }
 
+def _enrich_service(client, cluster_arn, service_arn):
+    service = describe_service(client, cluster_arn, service_arn)
 
-def get_service_list(ecs_client, cluster_arn):
-    service_list = []
+    task_definitions = [
+        _describe_task_definition(client, deployment["taskDefinition"])
+        for deployment in service["deployments"]
+    ]
 
-    for service_arn in get_service_arns(ecs_client, cluster_arn):
-        service = describe_service(ecs_client, cluster_arn, service_arn)
-        service_list.append(_create_service_dict(service))
+    zipped = zip(service["deployments"], task_definitions)
 
-    return service_list
+    enriched_deployments = [
+        _enrich_deployment(deployment, task_definition)
+        for deployment, task_definition in zipped
+    ]
 
+    enriched_events = [_create_event_dict(e) for e in service["events"][:5]]
```

Comment:
Why are you only getting the first five events?

Reply:
Because there could be _a lot_ of them (and only the recent ones are useful for the dashboard).

Reply:
Ah, I see the original comment – can we keep that please? :-)